ctapipe/atmosphere.py

```diff
@@ -0,0 +1,389 @@
+#!/usr/bin/env python3
+
+"""Atmosphere density models and functions to transform between column density
+(X in grammage units) and height (meters) units.
+
+Zenith angle is taken into account in the line-of-sight integral to compute the
+column density X assuming Earth as a flat plane (the curvature is not taken into
+account)
+
+"""
+
+import abc
+from dataclasses import dataclass
+from functools import partial
+from typing import Dict
+
+import numpy as np
+from astropy import units as u
+from astropy.table import Table
+from scipy.interpolate import interp1d
+
+__all__ = [
+    "AtmosphereDensityProfile",
+    "ExponentialAtmosphereDensityProfile",
+    "TableAtmosphereDensityProfile",
+    "FiveLayerAtmosphereDensityProfile",
+]
+
+
+class AtmosphereDensityProfile:
+    """
+    Base class for models of atmosphere density.
+    """
+
+    @abc.abstractmethod
+    def __call__(self, height: u.Quantity) -> u.Quantity:
+        """
+        Returns
+        -------
+        u.Quantity["g cm-3"]
+            the density at height h
+        """
+        raise NotImplementedError()
```

Comment:
not really needed, this will never be called. Just having the docstring is fine (and won't result in codecov complaining)